parsers/javascript/CharIterator.js

```diff
@@ -0,0 +1,29 @@
+export default class CharIterator {
+  #chars;
```

Comment:
TIL public and private class fields now have wide support in evergreen browsers. Nice!